test/specs/lib/FocusZone-test.tsx

```diff
@@ -0,0 +1,1451 @@
+import * as React from 'react'
+
+import * as ReactDOM from 'react-dom'
+import * as ReactTestUtils from 'react-dom/test-utils'
+import { KeyCodes } from '@uifabric/utilities'
+
+import {
+  FocusZone,
+  FocusZoneDirection,
+  FocusZoneTabbableElements,
+} from '../../../src/lib/accessibility/FocusZone'
+import { IS_FOCUSABLE_ATTRIBUTE } from 'src/lib/accessibility/interfaces'
+
+describe('FocusZone', () => {
+  let lastFocusedElement: HTMLElement | undefined
+  function onFocus(ev: any): void {
+    lastFocusedElement = ev.target
+  }
+
+  function setupElement(
+    element: HTMLElement,
+    {
+      clientRect,
+      isVisible = true,
+    }: {
+      clientRect: {
+        top: number
+        left: number
+        bottom: number
+        right: number
+      }
+      isVisible?: boolean
+    },
+  ): void {
+    element.getBoundingClientRect = () => ({
+      top: clientRect.top,
+      left: clientRect.left,
+      bottom: clientRect.bottom,
+      right: clientRect.right,
+      width: clientRect.right - clientRect.left,
+      height: clientRect.bottom - clientRect.top,
+    })
+
+    element.setAttribute('data-is-visible', String(isVisible))
+
+    element.focus = () => ReactTestUtils.Simulate.focus(element)
+  }
+
+  beforeEach(() => {
+    lastFocusedElement = undefined
+  })
+
+  it('can use arrows vertically', () => {
+    const component = ReactTestUtils.renderIntoDocument(
+      <div {...{ onFocusCapture: onFocus }}>
+        <FocusZone direction={FocusZoneDirection.vertical}>
+          <button className="a">a</button>
+          <button className="b">b</button>
+          <button className="c">c</button>
+        </FocusZone>
+      </div>,
+    )
+
+    const focusZone = ReactDOM.findDOMNode(component as React.ReactInstance)!!.firstChild as Element
+
+    const buttonA = focusZone.querySelector('.a') as HTMLElement
+    const buttonB = focusZone.querySelector('.b') as HTMLElement
+    const buttonC = focusZone.querySelector('.c') as HTMLElement
+
+    // Assign bounding locations to buttons.
+    setupElement(buttonA, {
+      clientRect: {
+        top: 0,
+        bottom: 30,
+        left: 0,
+        right: 100,
+      },
+    })
+
+    setupElement(buttonB, {
+      clientRect: {
+        top: 30,
+        bottom: 60,
+        left: 0,
+        right: 100,
+      },
+    })
+
+    setupElement(buttonC, {
+      clientRect: {
+        top: 60,
+        bottom: 90,
+        left: 0,
+        right: 100,
+      },
+    })
+
+    // Focus the first button.
+    ReactTestUtils.Simulate.focus(buttonA)
+    expect(lastFocusedElement).toBe(buttonA)
+
+    // Pressing down should go to b.
+    ReactTestUtils.Simulate.keyDown(focusZone, { which: KeyCodes.down })
```

Comment:
Heads up, `event.which` is a [deprecated](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/which) value for keyboard events.  See the [`keyboard-key`](https://www.npmjs.com/package/keyboard-key) README for more info.

Reply:
Nice work with this npm package! Unfortunately, ReactTestUtils do not work with `key`, I just tested that and it seems like [I am not the only one](https://stackoverflow.com/questions/30628093/reacts-testutils-simulate-keydown-does-not-work). This means that we should probably keep it as it is now and maybe consider creating a bug report for that in their repo.

I at least changed all the expects to use `keyboardKey.getCode` which works. 👍